Tests/GeoLib/TestLineSegmentIntersect2d.cpp

```diff
@@ -50,18 +50,29 @@ class LineSegmentIntersect2dTest : public testing::Test
 	ac::gtest_reporter gtest_reporter;
 };
 
+#if not defined(_MSC_VER) && _MSC_VER < 1900
+// Compilers of MVS below 2015 do not support unrestricted unions. The
+// unrestricted union is used by autocheck to handle test data. The autocheck
+// workaround for MVS compilers (below version 2015) contains a bug and in the
+// consequence the tests crashes. For this reason the tests are disabled under
+// this environments.
```

Comment:
But it at least compiles under old MSVC++, right?
Then I'd prefer not to `#if` out the whole test, but to do
```
#if ...
TEST_F(LineSegmentIntersect2dTest, RandomSegmentOrientationIntersecting)
#else
TEST_F(LineSegmentIntersect2dTest, RandomSegmentOrientationIntersecting_DISABLED)
#endif
```
That's how we treat non-working PETSc tests, too.

Maybe you could also add a comment on the experienced error message, either in the code or in the discussion of the PR, because it took some time to figure out what the error was and that it was not your/pur fault.